src/server.h

```diff
@@ -3154,6 +3154,20 @@ robj *setTypeDup(robj *o);
 #define HASH_SET_TAKE_VALUE (1<<1)
 #define HASH_SET_COPY 0
 
+/* Each dict of hash object that has fields with time-Expiration will have the
+ * following metadata attached to dict header */
+typedef struct dictExpireMetadata {
+    ExpireMeta expireMeta;   /* embedded ExpireMeta in dict.
+                                To be used in order to register the hash in the
+                                global ebuckets (i.e db->hexpires) with next,
+                                minimum, hash-field to expire */
+    ebuckets hfe;            /* DS of Hash Fields Expiration, associated to each hash */
+    sds key;                 /* reference to the key, same one that stored in
+                               db->dict. Will be used from active-expiration flow
+                               for notification and deletion of the object, if
+                               needed. */
+} dictExpireMetadata;
+
```

Comment:
Not sure if it worth to expose this structure and function `isDictWithMetaHFE()`. 
A simplified alternative my be extend t_hash.c with specific API for your need.

Reply:
if so, we need to add at least three method like `hashTypeHasMetaHFE()`, `HashTypeSetMetaKey()`, `HashTypeGetHFE()`.